### PR TITLE
No completion on attributes with xs:union (XSD)

### PR DESCRIPTION
**Pull request by angelozerr**
No completion on attributes with xs:union (XSD)

Fixes https://github.com/redhat-developer/vscode-xml/issues/351

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR 

 * fixes redhat-developer/vscode-xml#351
 * manage documentation from xs:enumeration inside attribute value
 * manage documentation from xs:enumeration inside text node

To play with this PR, you can write the `dressSize.xsd` file:

```
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
	elementFormDefault="qualified" attributeFormDefault="unqualified">

	<xs:simpleType name="SizeType">
		<xs:union memberTypes="DressSizeType">
			<xs:annotation>
				<xs:documentation>Size Type documentation</xs:documentation>
			</xs:annotation>
			<xs:simpleType>
				<xs:restriction base="xs:token">
					<xs:enumeration value="small">
						<xs:annotation>
							<xs:documentation>Small documentation</xs:documentation>
						</xs:annotation>
					</xs:enumeration>
					<xs:enumeration value="medium">
						<xs:annotation>
							<xs:documentation>Medium documentation</xs:documentation>
						</xs:annotation>
					</xs:enumeration>
					<xs:enumeration value="large" />
				</xs:restriction>
			</xs:simpleType>
		</xs:union>
	</xs:simpleType>
	<xs:simpleType name="DressSizeType">
		<xs:restriction base="xs:integer">
			<xs:minInclusive value="2" />
			<xs:maxInclusive value="18" />
		</xs:restriction>
	</xs:simpleType>

	<xs:element name="dresssize" type="SizeType" />

	<xs:element name="dress">
		<xs:complexType>
			<xs:attribute name="size" type="SizeType" />
		</xs:complexType>
	</xs:element>

</xs:schema>
```

 * for testing xs:enumeration completion and hover inside attribute value, write this XML and play with size attribute value:

```xml
<dress
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:noNamespaceSchemaLocation="./dressSize.xsd"
	size="large" />
```

 * for testing xs:enumeration completion and hover inside text node, write this XML:
```xml
<dresssize
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:noNamespaceSchemaLocation="./dressSize.xsd">
	
	small
	</dresssize>
```

**Comment by datho7561** [comment]
If a blank `<xs:documentation>` is provided under enum, then instead of showing the documentation for the parent type, it doesn't show any documentation. It might be worthwhile to show the parent documentation in this case if it easy to do so.

To reproduce:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
	elementFormDefault="qualified" attributeFormDefault="unqualified">

	<xs:simpleType name="SizeType">
		<xs:union memberTypes="DressSizeType">
			<xs:annotation>
				<xs:documentation>Size Type documentation</xs:documentation>
			</xs:annotation>
			<xs:simpleType>
				<xs:restriction base="xs:token">
					<xs:enumeration value="small">
						<xs:annotation>
							<xs:documentation></xs:documentation>
						</xs:annotation>
					</xs:enumeration>
					<xs:enumeration value="medium">
						<xs:annotation>
							<xs:documentation>Medium documentation</xs:documentation>
						</xs:annotation>
					</xs:enumeration>
					<xs:enumeration value="large" />
				</xs:restriction>
			</xs:simpleType>
		</xs:union>
	</xs:simpleType>
	<xs:simpleType name="DressSizeType">
		<xs:restriction base="xs:integer">
			<xs:minInclusive value="2" />
			<xs:maxInclusive value="18" />
		</xs:restriction>
	</xs:simpleType>

	<xs:element name="dresssize" type="SizeType" />

	<xs:element name="dress">
		<xs:complexType>
			<xs:attribute name="size" type="SizeType" />
		</xs:complexType>
	</xs:element>

</xs:schema>
```

Make an XML that associates with the above schema, then open the completion for a dresssize and select `small`. No documentation is shown.

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/dtd/contentmodel/CMDTDAttributeDeclaration.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public boolean isRequired() {
 	}
 
 	@Override
-	public String getValueDocumentation(String value, ISharedSettingsRequest settings) {
+	public String getAttributeValueDocumentation(String value, ISharedSettingsRequest settings) {
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/dtd/contentmodel/CMDTDAttributeDeclaration.java]: fixed

**Comment by angelozerr** [comment]
> Make an XML that associates with the above schema, then open the completion for a dresssize and select small. No documentation is shown.

It should be fixed and I wrote test for this usecase.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaHoverExtensionsTest.java [hunk 2]
```diff
@@ -227,6 +227,37 @@ public void hoverCacheBug() throws BadLocationException, MalformedURIException {
 
 	}
 
+	@Test
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaHoverExtensionsTest.java:230]: Yes it's intentional. If attribute value documentation cannot be found we display attribute name documentation.

**Review by datho7561 — APPROVED** [review]

**Comment by angelozerr** [comment]
Thanks @datho7561 for your review!

**Pull request merged**